src/nemos/basis.py

```diff
@@ -1177,6 +1179,96 @@ def evaluate_on_grid(self, n_samples: int) -> Tuple[NDArray, NDArray]:
         return super().evaluate_on_grid(n_samples)
 
 
+class FourierBasis(Basis):
+    """Set of 1D Fourier basis.
+
+    This class defines a cosine and negative sine basis (quadrature pair)
+    with frequencies ranging 0 to max_freq.
+
+    Parameters
+    ----------
+    max_freq
+            Highest frequency of the cosine, negative sine pairs.
+            The number of basis function will be 2*max_freq + 1.
+    """
+
+    def __init__(self, max_freq: int):
+        super().__init__(n_basis_funcs=2 * max_freq + 1)
+
+        self._frequencies = np.arange(max_freq + 1, dtype=float)
+        self._n_input_dimensionality = 1
+
+    def _check_n_basis_min(self) -> None:
+        """Check that the user required enough basis elements.
+
+        Checks that the number of basis is at least 1.
+
+        Raises
+        ------
+        ValueError
+            If an insufficient number of basis element is requested for the basis type.
+        """
+        if self.n_basis_funcs < 0:
+            raise ValueError(
+                f"Object class {self.__class__.__name__} requires >= 1 basis elements. "
+                f"{self.n_basis_funcs} basis elements specified instead"
+            )
+
+    def evaluate(self, sample_pts: ArrayLike) -> NDArray:
+        """Generate basis functions with given spacing.
+
+        Parameters
+        ----------
+        sample_pts
+            Spacing for basis functions.
+
+        Returns
+        -------
+        basis_funcs
+            Evaluated Fourier basis, shape (n_samples, n_basis_funcs).
+
+        Notes
+        -----
+        The frequencies are set to np.arange(max_freq+1), convolving a signal
+        of length n_samples with this basis is equivalent, but slower,
+        then computing the FFT truncated to the first max_freq components.
+
+        Therefore, convolving a signal with this basis is equivalent
+        to compute the FFT over a sliding window.
+
+        Examples
+        --------
+        >>> import nemos as nmo
+        >>> import numpy as np
+        >>> n_samples, max_freq = 1000, 10
+        >>> basis = nmo.basis.FourierBasis(max_freq)
+        >>> eval_basis = basis.evaluate(np.linspace(0, 1, n_samples))
+        >>> sinusoid = np.cos(3 * np.arange(0, 1000) * np.pi * 2 / 1000.)
+        >>> conv = [np.convolve(eval_basis[::-1, k], sinusoid, mode='valid')[0] for k in range(2*max_freq+1)]
+        >>> fft = np.fft.fft(sinusoid)
+        >>> print('FFT power:   ', np.round(np.real(fft[:max_freq]), 4))
+        >>> print('Convolution: ', np.round(conv[:max_freq], 4))
```

Comment:
Is there a way in mkdocs to set this as a python codeblock? Can you remove the `>>>`? Because right now in the docs the `>>>` shows up and when you copy it copies with the `>>>` so a user can't just copy and paste and run from the docs :smile: 

for example at the bottom here the arrows render and get copied: https://nemos.readthedocs.io/en/latest/reference/nemos/utils/#nemos.utils.pytree_map_and_reduce